src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPlugin.java

```diff
@@ -1,46 +1,299 @@
 
-    private String region;
-
-    public AwsV2NetworkPlugin(String confFilePath) {
-        Properties properties = PropertiesUtil.readProperties(confFilePath);
-        this.region = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_REGION_SELECTION_KEY);
-    }
-
-    @Override
-    public String requestInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
-    	throw new UnsupportedOperationException("This feature has not been implemented for aws cloud, yet.");
-    }
-
-    @Override
-    public NetworkInstance getInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
-    	throw new UnsupportedOperationException("This feature has not been implemented for aws cloud, yet.");
-    }
-
-    @Override
-    public void deleteInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
-    	throw new UnsupportedOperationException("This feature has not been implemented for aws cloud, yet.");
-    }
-
-    @Override
-    public boolean isReady(String instanceState) {
-        return false;
-    }
-
-    @Override
-    public boolean hasFailed(String instanceState) {
-        return true;
-    }
+	private static final Logger LOGGER = Logger.getLogger(AwsV2NetworkPlugin.class);
+
+	private static final String ALL_PROTOCOLS = "-1";
+	private static final String AWS_TAG_GROUP_ID = "groupId";
+	private static final String AWS_TAG_NAME = "Name";
+	private static final String LOCAL_GATEWAY_DESTINATION = "local";
+	private static final String SECURITY_GROUP_DESCRIPTION = "Security group associated with a fogbow network.";
+	private static final Object SECURITY_GROUP_RESOURCE = "Security Groups";
+	private static final String SUBNET_RESOURCE = "Subnet";
+
+	private String region;
+	private String vpcId;
+	private String zone;
+
+	public AwsV2NetworkPlugin(String confFilePath) {
+		Properties properties = PropertiesUtil.readProperties(confFilePath);
+		this.region = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_REGION_SELECTION_KEY);
+		this.vpcId = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_DEFAULT_VPC_ID_KEY);
+		this.zone = properties.getProperty(AwsV2ConfigurationPropertyKeys.AWS_AVAILABILITY_ZONE_KEY);
+	}
+
+	@Override
+	public String requestInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.REQUESTING_INSTANCE, cloudUser.getToken()));
+
+		String cidr = networkOrder.getCidr();
+		String name = defineInstanceName(networkOrder.getName());
+
+		CreateSubnetRequest request = CreateSubnetRequest.builder()
+				.availabilityZone(this.zone)
+				.cidrBlock(cidr)
+				.vpcId(this.vpcId)
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		String subnetId = doCreateSubnetResquests(name, request, client);
+		doCreateSecurityGroupRequests(cidr, subnetId, client);
```

Comment:
I think it could be named as handleSecurityIssues or something like that, once in this method u create the group and add the rules...

Reply:
I liked the suggestion, I'll use it.